jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java

```diff
@@ -486,9 +497,41 @@ public static String getEnvVarOrSystemProperty(String envVarName, String systemP
         if (StringUtils.isNotBlank(ret)){
             return ret;
         }
-        return System.getProperty(systemProperty, defaultValue);
+        return Optional.ofNullable(getProperty(systemProperty)).orElse(defaultValue);
     }
 
-}
+    /**
+     * Utility method to check whether underlying processor architecture is ARM or not
+     * @return boolean value which is true only when underlying processor architectureis ARM
+     */
+    public static boolean isProcessorArchitectureARM() {
+        return getEnvVarOrSystemProperty("", "os.arch", "amd64").contains("aarch");
+    }
```

Comment:
Why don't we just provide a util to get osArch and then do any logic downstream.
Also, again you're using contains which is quite dangerous (less in this case because the number and names of platforms are limited)